lib/mix/lib/mix/dep/loader.ex

```diff
@@ -36,6 +36,15 @@ defmodule Mix.Dep.Loader do
     only != nil and env not in List.wrap(only)
   end
 
+  def with_env_vars(%Mix.Dep{env_vars: env_vars}, callback) do
```

Comment:
Let's have a fast path clause that checks if env_vars is an empty list and then we just invoke the callback.